modules/imgcodecs/test/test_read_write.cpp

```diff
@@ -303,4 +303,75 @@ TEST(Imgcodecs_Image, write_umat)
     EXPECT_EQ(0, remove(dst_name.c_str()));
 }
 
+TEST(Imgcodecs_Image, multipage_collection_size)
+{
+    const string src_name = TS::ptr()->get_data_path() + "readwrite/multipage.tif";
+    ImageCollection collection(src_name, IMREAD_ANYCOLOR);
+    EXPECT_EQ((size_t)6, collection.size());
+}
+
+TEST(Imgcodecs_Image, multipage_collection_iterator)
+{
+    const string src_name = TS::ptr()->get_data_path() + "readwrite/multipage.tif";
+    ImageCollection collection(src_name, IMREAD_ANYCOLOR);
+
+    int count = 0;
+    auto prev = collection.begin().operator*();
+
+    // Test if each page is different from each other. In the multipage.tif file each page is different.
+    for(auto&& i : collection) {
+        // pass first image
+        if(count == 0) {
+            count++;
+            continue;
+        }
+        auto curr = i;
+        bool isEqual = (sum(prev != curr) == Scalar(0,0,0,0));
```

Comment:
when all pages are the same, it's quite bad test. We need something different. I would create a multi-page image with different pages, then stored each page as individual file and then compared each page of multi-page image with the same page read as individual image, e.g.:
```
ImageCollection imcol("multi-page.tiff");
int i = 0;
for (auto& page : imcol) {
     double diff = norm(page, imread(format("multi-page-%d.tiff", i), NORM_INF);
     EXPECT_EQ(diff, 0.);
     i++;
}
```